be/src/vec/functions/function_map.cpp

```diff
@@ -440,13 +518,409 @@ class FunctionStrToMap : public IFunction {
+                                                         map_nullmap);
+        case TYPE_DATE:
+            return _execute_dispatch_value<ColumnDate>(value_type, map_column, key_column,
+                                                       key_nullmap, value_column, value_nullmap,
+                                                       map_nullmap);
+        case TYPE_DATETIME:
+            return _execute_dispatch_value<ColumnDateTime>(value_type, map_column, key_column,
+                                                           key_nullmap, value_column, value_nullmap,
+                                                           map_nullmap);
+        case TYPE_DATEV2:
+            return _execute_dispatch_value<ColumnDateV2>(value_type, map_column, key_column,
+                                                         key_nullmap, value_column, value_nullmap,
+                                                         map_nullmap);
+        case TYPE_DATETIMEV2:
+            return _execute_dispatch_value<ColumnDateTimeV2>(value_type, map_column, key_column,
+                                                             key_nullmap, value_column,
+                                                             value_nullmap, map_nullmap);
+        case TYPE_TIMEV2:
+            return _execute_dispatch_value<ColumnTimeV2>(value_type, map_column, key_column,
+                                                         key_nullmap, value_column, value_nullmap,
+                                                         map_nullmap);
+        default:
+            return nullptr;
+        }
+    }
+
+    // type comparability check and dispatch
+    Status _execute_type_check_and_dispatch(Block& block, const ColumnNumbers& arguments,
+                                            uint32_t result, size_t input_rows_count) const {
+        // extract map column
+        auto map_column_ptr =
+                block.get_by_position(arguments[0]).column->convert_to_full_column_if_const();
+        const ColumnMap* map_column = nullptr;
+        const UInt8* map_nullmap = nullptr;
+        if (map_column_ptr->is_nullable()) {
+            const auto* nullable_column =
+                    reinterpret_cast<const ColumnNullable*>(map_column_ptr.get());
+            map_column = check_and_get_column<ColumnMap>(nullable_column->get_nested_column());
+            map_nullmap = nullable_column->get_null_map_column().get_data().data();
+        } else {
+            map_column = check_and_get_column<ColumnMap>(*map_column_ptr.get());
+        }
+        if (!map_column) {
+            return Status::RuntimeError("unsupported types for function {}({})", get_name(),
+                                        block.get_by_position(arguments[0]).type->get_name());
+        }
+
+        // extract (search) key and value columns
+        auto key_column_ptr =
+                block.get_by_position(arguments[1]).column->convert_to_full_column_if_const();
```

Comment:
Maybe should try to use `unpack_if_const` instead of `convert_to_full_column_if_const`.